src/js/components/DataTable/DataTable.js

```diff
@@ -151,6 +151,43 @@ const DataTable = ({
   // offset compensation when body overflows
   const [scrollOffset, setScrollOffset] = useState(0);
 
+  // multiple pinned columns offset
+  const [pinnedOffset, setPinnedOffset] = useState();
+
+  const getPinnedOffset = useCallback(() => {
+    const pinnedProperties = columns
+      .map(pinnedColumn => pinnedColumn.pin && pinnedColumn.property)
+      .filter(n => n);
+
+    const pinnedColumnsProperties = {};
+    const firstRowCells = headerRef.current.firstChild.cells || [];
```

Comment:
This is feeling like DataTable is reaching inside Header a bit too much. I'm wondering if we should move this inside Header and Cell, perhaps something like:
`<Header onWidths={([n,n,n]) => {}} />` and `<Cell onWidth={(width) => {}} />`
?

Reply:
I'm always open to suggestions, but I'm not sure if I'm following this one, sorry 😄 can you go a little further on that?

Reply:
So, rather than have DataTable know that the Header's first child contains cells, which is knowing more about the Header than DataTable should, have the DataTable pass an `onWidths` to Header, indicating that the DataTable wants to be told the Header cell widths. When `onWidths` is passed to Header, Header passes an `onWidth` to each Cell. When a Cell has an `onWidth`, it reports its width, possible via a `useLayoutEffect()`. Header merges these into an array that it passes back up to DataTable.

The motivation is to not have DataTable know about the internals of the DOM structure of Header, to separate concerns. I am thinking this is just a few lines in each file. But, if going this route appears to be too complicated, I'm willing to consider other approaches.